src/torrent.cpp

```diff
@@ -2100,9 +2101,10 @@ namespace libtorrent
 		inc_refcount("check_fastresume");
 		// async_check_fastresume will gut links
 		m_ses.disk_thread().async_check_fastresume(
-			m_storage.get(), m_resume_data ? &m_resume_data->node : NULL
+			m_storage.get(), m_resume_data && !m_check_fastresume_queued ? &m_resume_data->node : NULL
```

Comment:
I'm a bit hesitant to add another piece of state. if we no longer need the resume data in the torrent, maybe it could be moved into the job somehow. ideally without being copied. it's held by a scoped_ptr now. it could be changed to a unique_ptr, and it could be released here, and captured again in the disk job.